### PR TITLE
Feature/project structure

### PR DESCRIPTION
**Pull request by Camilo-Camargo**
# Project Structure

## Description
The pull request establishes the project structure based on the following libraries
- [sqlc](https://sqlc.dev/):  Generate all the repository operations based on SQL
- [golang-migrate](https://github.com/golang-migrate/migrate): Create the migrations based on the schemas created to use in conjunction sqlc.
- [fiber](https://gofiber.io/): An express like web framework.
- [validator](https://github.com/go-playground/validator): Validation to structs.

### Summary
Create a project structure using the most efficient and productive libraries.

## Project Structure
The project is structure in the following directories
```bash
.
├── README.md
├── cmd # All the programs
│   ├── cli
│   └── server
├── compose-dev.yml
├── database # Stores the database information
│   ├── db.go
│   ├── migrate.go
│   ├── queries # Contains the repository queries
│   ├── schemas # Contains the SQL entity definitions
│   ├── seeder.go
│   └── seeders # Contains the seeders
├── go.mod
├── go.sum
├── internal
│   ├── dto # Contains all the request and response contracts used by services
│   ├── handler # Contains all the controllers used by the routes
│   ├── repository # Contains the SQLC-generated code
│   ├── router # Defines the exposed endpoints
│   ├── service # Defines all the business services that connect with the repository
│   └── sqlc
├── scripts # Contains the scripts needed to run the project
└── sqlc.yaml
```

To run the project structure, you must follow the `README.md`

## How Has This Been Tested?
Describe the tests that you ran to verify your changes. Provide instructions so others can reproduce. Please also list any relevant details for your test configuration.

- **TestClientServices:** This test uses CreateClientService to create a user in the database and check if the `ID` is the same as the GetClientSrevice.

### Test Configuration:
- **Docker**:*version 25.0.3, build 4debf41* 
- **Docker Compose**:*Docker Compose version v2.24.5-desktop.1* 
- **OS**:*MacOs*
- **Version**:*Sonoma 14.4.1*

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Alejandro-Suarez-Rincon — APPROVED** [review]

File: database/db.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+var instance *sql.DB
+
+func DBUrl() string {
+	dbUser := os.Getenv("DATABASE_USER")
+	dbPass := os.Getenv("DATABASE_PASSWORD")
+	dbHost := os.Getenv("DATABASE_HOST")
+	dbName := os.Getenv("DATABASE_NAME")
+	dbPort := os.Getenv("DATABASE_PORT")
+
```
Comment (JosephSC0121) [inline database/db.go:19]: qué tal agregar una validación aqui por si algún dato no se leyó del .env? puede ser un TODO para luego

File: database/db.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	return fmt.Sprintf("host=%s port=%s user=%s password=%s dbname=%s sslmode=disable", dbHost, dbPort, dbUser, dbPass, dbName)
```
Comment (JosephSC0121) [inline database/db.go:20]: Otra cosa, no puede ser peligroso imprimir la password de la db? XD

File: internal/router/routes.go [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package router
+
+import (
+	"optitech/internal/handler"
+)
+
```
Comment (JosephSC0121) [inline internal/router/routes.go:6]: variables un poco más descriptivas :D

File: internal/router/routes.go [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+func (s Server) AttachRoutes() {
+	r := s.app
```
Comment (JosephSC0121) [inline internal/router/routes.go:8]: Variables un poco más descriptivas

File: internal/service/client_test.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+		getClient, err = GetClientService(req)
+		assert.NotNil(t, getClient)
+		assert.Nil(t, err)
+	})
+
+	assert.NotNil(t, getClient)
+	assert.NotNil(t, client)
+	assert.Equal(t, getClient.Id, client.ClientID)
+}
```
Comment (JosephSC0121) [inline internal/service/client_test.go:53]: genial!!

**Review by JosephSC0121 — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -1 +1,26 @@
...
+<h1 align="center">
+    optitech API
+</h1>
+
+## Getting Started
+To start the project in development mode, you must copy the `.env.example` to `.env`:
+and then you should run the following command:
+
+```bash
+docker compose -f compose-dev.yaml
```
Comment (JosephSC0121) [inline README.md:10]:
```suggestion
docker compose -f compose-dev.yml up
```

**Review by CamiloArias12 — APPROVED** [review]

File: internal/router/routes.go (hunk 4 above)
Comment (Camilo-Camargo) [reply internal/router/routes.go:8]: https://go.dev/wiki/CodeReviewComments#variable-names

**Pull request merged**